clarity/src/vm/contracts.rs

```diff
@@ -35,19 +38,28 @@ pub struct Contract {
 impl Contract {
     pub fn initialize_from_ast(
         contract_identifier: QualifiedContractIdentifier,
-        contract: &ContractAST,
+        contract: &mut ContractAST,
+        contract_analysis: &ContractAnalysis,
         sponsor: Option<PrincipalData>,
         global_context: &mut GlobalContext,
         version: ClarityVersion,
     ) -> Result<Contract> {
         let mut contract_context = ContractContext::new(contract_identifier, version);
 
-        eval_all(
-            &contract.expressions,
-            &mut contract_context,
-            global_context,
-            sponsor,
-        )?;
+        if let Some(wasm_module) = contract.wasm_module.take() {
+            contract_context.set_wasm_module(wasm_module);
+
+            // Execute the contract via the compiled Wasm module
+            initialize_contract(global_context, &mut contract_context, contract_analysis)?;
```

Comment:
Hm, in this context `initialize_contract` doesn't really feel like the right naming, since it's actually executing (according to the comment)?

Reply:
It's "initializing the contract" in that it's running the top-level code, that only gets run once, during deployment, so it's initialization code. I can improve the comment to make that more clear.

Reply:
Added comments in ce8488c55